radixdlt-core/radixdlt/src/main/java/com/radixdlt/client/store/berkeley/StackingCollector.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * (C) Copyright 2021 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.client.store.berkeley;
+
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+/**
+ * The purpose of this data structure is to collect elements (using {@link #push(Object)} method) and then
+ * process collected elements at once using {@link #consumeCollected(Consumer)} method. While processing takes place
+ * new elements can be stored (they will be processed during next call to {@link #consumeCollected(Consumer)}).
+ */
+public final class StackingCollector<T> {
```

Comment:
Perhaps better to use RxJava3 primitives to handle async boundary logic. Would then be able to integrate with our current dispatcher framework.

Reply:
This class is just a thread-safe temporary storage.